diff --git a/src/extract_gpuinfo_amdgpu.c b/src/extract_gpuinfo_amdgpu.c
index beb6a99..78ae3f7 100644
--- a/src/extract_gpuinfo_amdgpu.c
+++ b/src/extract_gpuinfo_amdgpu.c
@@ -35,7 +35,14 @@
 #include <inttypes.h>
 #include <libdrm/amdgpu.h>
 #include <libdrm/amdgpu_drm.h>
-#include <linux/kcmp.h>
+// Workaround for compiling with older kernel modules
+// Without the appropriate kernel the syscall will likely just fail
+// These users likely don't even have AMD GPU
+// #include <linux/kcmp.h>
+// https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/unistd.h#L733
+#define SYS_kcmp 272
+// https://github.com/torvalds/linux/blob/master/include/uapi/linux/kcmp.h#L8
+#define KCMP_FILE 0
 #include <math.h>
 #include <stdarg.h>
 #include <stdbool.h>
diff --git a/src/extract_processinfo_fdinfo.c b/src/extract_processinfo_fdinfo.c
index 636d517..bb60e92 100644
--- a/src/extract_processinfo_fdinfo.c
+++ b/src/extract_processinfo_fdinfo.c
@@ -25,7 +25,14 @@
 #include <ctype.h>
 #include <dirent.h>
 #include <fcntl.h>
-#include <linux/kcmp.h>
+// Workaround for compiling with older kernel modules
+// Without the appropriate kernel the syscall will likely just fail
+// These users likely don't even have AMD GPU
+// #include <linux/kcmp.h>
+// https://github.com/torvalds/linux/blob/master/include/uapi/asm-generic/unistd.h#L733
+#define SYS_kcmp 272
+// https://github.com/torvalds/linux/blob/master/include/uapi/linux/kcmp.h#L8
+#define KCMP_FILE 0
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/syscall.h>
